s3path/current_version.py

```diff
@@ -88,21 +89,21 @@ def from_uri(cls, uri: str):
         return cls(unquoted_uri[4:])
 
     @classmethod
-    def from_bucket_key(cls, bucket: str, key: str):
+    def from_bucket_key(cls, bucket: str, key: str) -> Self:
         """
         from_bucket_key class method create a class instance from bucket, key pair's
 
         >> from s3path import PureS3Path
         >> PureS3Path.from_bucket_key(bucket='<bucket>', key='<key>')
         << PureS3Path('/<bucket>/<key>')
         """
-        bucket = cls(cls.parser.sep, bucket)
-        if len(bucket.parts) != 2:
-            raise ValueError(f'bucket argument contains more then one path element: {bucket}')
-        key = cls(key)
-        if key.is_absolute():
-            key = key.relative_to('/')
-        return bucket / key
+        bucket_path = cls(cls.parser.sep, bucket)
+        if len(bucket_path.parts) != 2:
+            raise ValueError(f'bucket argument contains more then one path element: {bucket_path}')
+        key_path = cls(key)
+        if key_path.is_absolute():
+            key_path = key_path.relative_to('/')
+        return bucket_path / key_path
```

Comment:
This doesn't change any behavior, it just uses a new variable to prevent type confusion from overwriting a parameter with a different type.